pkg/result/processors/path_relativity.go

```diff
@@ -0,0 +1,60 @@
+package processors
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/golangci/golangci-lint/pkg/fsutils"
+	"github.com/golangci/golangci-lint/pkg/logutils"
+	"github.com/golangci/golangci-lint/pkg/result"
+)
+
+var _ Processor = (*PathRelativity)(nil)
+
+// PathRelativity computes [result.Issue.RelativePath] and [result.Issue.WorkingDirectoryRelativePath],
+// based on the base path.
+type PathRelativity struct {
+	log      logutils.Log
+	basePath string
+	wd       string
```

Comment:
personal pref: I prefer avoiding abbreviations like this on struct fields since having the full name avoids any ambiguity and makes it clearer. Don't seem like we refer to it an awful lot to make it an issue in the code.

```suggestion
	workingDirectory       string
```